angularjs-portal-home/src/main/webapp/my-app/marketplace/controllers.js

```diff
@@ -2,13 +2,15 @@
 
 define(['angular', 'jquery', 'require'], function(angular, $, require) {
 
-    var app = angular.module('my-app.marketplace.controllers', []);
+  var app = angular.module('my-app.marketplace.controllers', []);
 
-    app.controller('marketplaceCommonFunctions',
-      ['googleCustomSearchService', 'miscSearchService', 'layoutService', 'marketplaceService', 'miscService', 'MISC_URLS', '$sessionStorage',
-       '$localStorage','$rootScope', '$scope', '$routeParams', '$timeout', '$location',
-       function(googleCustomSearchService, miscSearchService, layoutService, marketplaceService, miscService,MISC_URLS, $sessionStorage,
-        $localStorage, $rootScope, $scope, $routeParams, $timeout, $location){
+  app.controller('marketplaceCommonFunctions', ['googleCustomSearchService', 'miscSearchService', 'layoutService', 'marketplaceService',
+    'miscService', 'MISC_URLS', '$sessionStorage', '$localStorage','$rootScope', '$scope', '$routeParams', '$timeout', '$location',
+    function(googleCustomSearchService, miscSearchService, layoutService, marketplaceService, miscService, MISC_URLS, $sessionStorage,
+             $localStorage, $rootScope, $scope, $routeParams, $timeout, $location) {
+
+      var currentThemePrimary = $sessionStorage.portal.theme.materialTheme.primary['500'];
+      $scope.primaryColorRgb = 'rgb('+ currentThemePrimary.value[0] + ',' + currentThemePrimary.value[1] + ',' + currentThemePrimary.value[2] + ')';
```

Comment:
I tried using $mdColors, but I couldn't get it to work without a prohibitively massive amount of work. I'm not concerned about the potential palette problem at the moment because every single theme has a custom palette. I could write a check to ensure there's a materialTheme defined (fall back on black if not). Would that be okay?